src/Tools/dotnet-monitor/Extensibility/FileProvider/FolderFileProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    /// <summary>
+    /// This class provides an implementation of <see cref="IFileProvider"/> with a root directory that does not need to exist at construction time.
+    /// </summary>
+    public class FolderFileProvider : IFileProvider
```

Comment:
Rather than having a new class that completely defers to either the `PhysicalFileProvider` or `MissingDirectoryProvider`, could we have a static method that determines this and returns the appropriate one?

Reply:
The reason for this class is for the `ExtensionRepository` to be initialized, it needs an `IFileProvider`, the `PhysicalFileProvider` checks upfront if the parent directory exists. This is undesirable, because we want the extension folder to be added after boot up; this class allows the folder to not exist.

Reply:
> because we want the extension folder to be added after boot up

Why? It's much easier to require that the extension folder exists before starting dotnet-monitor.

If we get into the business of allowing extensions to be added during dotnet-monitor lifetime, wouldn't you expect to be able to remove them as well? What if you delete the content of the extension folder and add new content? I don't think we want to have to deal with various "extension did or did not exist now, and may or may not exist later" scenarios.

Reply:
Sure, I got rid of this whole thing and just use the integrated physical providers as needed.